lib/EncodeImagesStage.js

```diff
@@ -4,20 +4,32 @@ var Promise = require('bluebird');
 
 Jimp.prototype.getBufferAsync = Promise.promisify(Jimp.prototype.getBuffer);
 
-module.exports = encodeImages;
+module.exports = EncodeImagesStage;
+
+/**
+ * Encodes Jimp images in pipeline extras to buffers.
+ *
+ * The glTF asset must be initialized for the pipeline.
+ */
+function EncodeImagesStage() {
+    this.name = 'encodeImages';
+    this.before = ['quantizeAttributes'];
+    this.after = [];
```

Comment:
The dependency graph is so, for example, if three stages all depend on stage `A`, stage `A` is only ran once, so it is required for reasonable performance.  It is OK to do it last, but it should be done as part of this PR.

There will also be some cases where we actually _don't_ want to optimize out a `before`/`after` stage and want it to run explicitly and immediately before/after the given stage.  @lasalvavida or @lilleyse should know the exact case(s) we currently have here.

For this, we probably need another set of arrays, `immediatelyBefore`/`immediatelyAfter` or whatever.  There, of course, will be some opportunities for optimization here where one stage might have stage `A` as immediate, and other stages have stage `A` as `before` (but not immediate), and stage `A` would only need to run once.